### DISCUSSION TITLE
[Projects beta] Automatically expand custom project fields in the issues view

### DISCUSSION THREAD

**Discussion by nkolev92** [community/community#15592] [Projects and Issues]
We use projects beta for sprint tracking etc. 

Often times when creating new issues we want to fill some custom fields, such as `Estimate`, or `Sprint` etc. 

It'd save a few click if when an issue is in a project, if the custom fields could be expanded by default. 

Currently it feels like a second class citizen compared to say `Milestone`.

**Comment by EbenezerJesuraj** [comment]
We use project fields to keep track of the changes happening over in the project..

For now we are able to create custom fields on a project level, but we also prefer either custom fields in the project pane also reflect on the issues view or atleast give us the option to create custom fields in the issues tab..

**Comment by ssbarnea** [comment]
Linked to https://github.com/refined-github/refined-github/issues/5825 but sadly refined-github does not want to have anything to do with projects.

I am in the same board, having only one field to display,... which is always hidden under the "more" part.

**Reply by archon810** [reply]
Well, if Github won't do it and Refined Github won't do it, someone else had to. https://github.com/orgs/community/discussions/15592#discussioncomment-17590561

**Comment by andrewle** [comment]
We were just bitten by this and would love for a config option for it to be automatically or always expanded as well

**Comment by NeemiasBorges** [comment]
I understand that these additions would really be very useful

**Comment by jeesim2** [comment]
`Projects beta` is already ready to prod I think. Also I agree the feature to expand it as default.

**Comment by calvinchankycc** [comment]
bumping this ticket. It would be nice to show my custom `sprint` field from github projects, in the ticket.

**Comment by otech47** [comment]
BUMP 

EXPAND PLEASE

**Comment by breathe** [comment]
If this discussion is about the rendering of the `Projects` component on the issues page.

![image](https://github.com/community/community/assets/434034/e997e402-d2f1-4718-85be-f5ec92144d1c)

Then I want to add my emphatic plus 1 to this issue ...

Notice that there's a +3 more there (you might not even notice it, took me awhile to realize it was there) -- and behind that is hiding the information/ui control that I would like to be shown all the time ...

![image](https://github.com/community/community/assets/434034/d9d69b15-0779-4561-8e64-0c75627e9ae4)

Not expanding that component by default saves exactly zero visual space on the page since this ui element is in its own column on the right with nothing meaningful below it anyway  ...)

**Comment by DitwanP** [comment]
BUMP! Would love to have it expanded by default.

**Comment by koppor** [comment]
We are starting to use projects more extensively. The expansion is really good to have an overview on the ticket. 

The other components on an issue are also always expanded:

- Reviewers
- Assignees
- Labels
- Development
- Participants

(Please keep them expanded 😅)

<details>
![image](https://github.com/user-attachments/assets/c4018fcb-7ca3-44d7-a0aa-25425dd38d4d)
</details>

**Comment by atis-s** [comment]
Please implement this! 
We miss lot of fields all the time because they are always hidden by default.

**Comment by koppor** [comment]
Please bring this to a discussion again. We are thinking of switching to labels again, because PR management with project boards at the side is not a nice UX.

**Comment by thisisthekap** [comment]
This teeny tiny change would help us a lot. Tripple Bump! :)

**Comment by labudis** [comment]
We shipped [Issue Fields](https://github.com/orgs/community/discussions/189141), which addresses this. Issue fields are org-level fields that appear directly in the issue sidebar, just like Assignees, Labels, and Milestone. No need to expand or navigate to a project to see them. You can pin specific fields to specific issue types, so only the relevant ones show up.

Issue fields are currently in public preview. Request access for your org in the [discussion](https://github.com/orgs/community/discussions/189141).

**Reply by atis-s** [reply]
Hi, sorry but how can this address the simple need for auto-expanded project fields? Many projects have separate fields that are not org-level fields.

**Reply by labudis** [reply]
It does not directly address the need of visualising the Project fields in the sidebar, but it allows you to move some of the fields into a more prominent position as long as they apply across the organisation. Just wanted to signpost the new capability in case it helps.

**Reply by archon810** [reply]
We have successfully moved Priority to the new org level issue field, though this had the side effect of losing priority for PRs. In fact, in general, not having support for these new issue fields in PRs, as well as the issue type, is quite limiting and a step back IMO. We should be allowed to decide where they show up and whether they apply to PRs as well.

@labudis Given that custom project fields are not going anywhere and they're not expanded by default requiring extra clicks, please add uncollapsing them, either by default or as a setting.

**Reply by archon810** [reply]
As a follow-up, I just discovered that the project fields are automatically expanded when an issue is opened from within the project view. It is only not expanded when opened on a dedicated issue page (or if it's a PR). So even GitHub itself doesn't have feature parity between the two views.

**Reply by MiloParigi** [reply]
Hi @labudis, while the new Issue Fields are great, I agree with the rest of the people here, it's quite limiting as we don't want client specific fields (SLOs/SLAs, client specific status, product affected, etc) to be shared with the rest of the organization. Is there any discussion on reopening this at Github ? It seems like a low hanging fruit to just add a settings controlling the default state (better yet would be to be able to select the fields that are shown by default, but I won't try to be more catholic than the Pope )

**Comment by archon810** [comment]
While we wait for an official solution from Github that may never come, I've come up with a GreaseMonkey/[TamperMonkey](https://www.tampermonkey.net/) script that automatically expands custom project fields for both issues and PRs. 

Version 1.0 is live here https://greasyfork.org/en/scripts/586383-github-auto-expand-project-fields.

If it works for people, I may turn it into a full Chrome extension eliminating the need for TamperMonkey.
